src/data_retrieval/__main__.py

```diff
@@ -22,98 +21,187 @@
-    output_path: str = os.path.expanduser("~/Downloads/ecmwf_global_forecast")
-    setup_output_path(output_path)
-
-    # Utilizing the specific range of available years you've provided
-    # Updated to include 1981 through 2023
-    available_years: List[int] = list(range(1981, 2024))
+def get_cds_ecmwf(local_path: Optional[str] = None, upload: bool = False):
+    logger.info("Downloading Copernicus CDS data of ECMWF global forecast...")
 
-    months: List[int] = list(range(1, 13))
+    # Define parameters 1981 to 2024 or 1982 to test
+    available_years = list(range(1981, 2024))
+    months = list(range(1, 13))
+    leadtime_months = list(range(1, 7))
 
-    # Adjusting for all available lead times
-    # Assuming these are all the lead times available
-    leadtime_months: List[int] = list(range(1, 7))
+    # Default path if no local_path is provided and not uploading
+    if not local_path and not upload:
+        local_path = os.path.expanduser("~/Downloads/ecmwf_global_forecast")
 
-    # Download all years, months, and leadtime months
-    download_ecmwf_cds(
-        available_years, months, leadtime_months, output_path, file_name
-    )
+    if local_path:
+        # Local download
+        file_name = "ecmwf_forecast_global_all_years.grib"
+        setup_output_path(local_path)
+        download_ecmwf_cds(
+            available_years, months, leadtime_months, local_path, file_name
+        )
+    elif upload:
+        sas_token, container_name, storage_account = load_env_vars()
+        # Upload to cloud, handle the stream
+        data_stream = download_ecmwf_cds(
+            available_years, months, leadtime_months
+        )
+        blob_path = (
+            "/raw/glb/ecmwf/ecmwf-monthly-seasonalforecast-1981-2023.grib"
+        )
+        upload_stream(
+            sas_token, container_name, storage_account, data_stream, blob_path
+        )
+    else:
+        logger.error(
+            "No valid operation specified. Please provide a local path or set upload to True."  # noqa: E501
+        )
 
 
-def get_cds_era5():
+def get_cds_era5(local_path: Optional[str] = None, upload: bool = False):
     logger.info(
         "Downloading Copernicus CDS data of ERA5 total precipitation.."
     )
-    file_name: str = (
-        "era5_total_precipitation_global_1981_2023_all_months.grib"
-    )
-    output_path: str = os.path.expanduser("~/Downloads/era5_global_data")
-    setup_output_path(output_path)
+    file_name = "era5_total_precipitation_global_1981_2023_all_months.grib"
 
     # Define the years and months for the single request
-    # From 1981 to 2023 try extra 2024 next time
-    years: List[int] = list(range(1981, 2024))
+    years = list(range(1981, 2024))
+    months = list(range(1, 13))
+
+    # Default path if no local_path is provided and not uploading
+    if not local_path and not upload:
+        local_path = os.path.expanduser("~/Downloads/era5_global_data")
+
+    if local_path and not upload:
```

Comment:
This is bug. Should be `elif` not `if`

When you run
```
poetry run python src/data_retrieval cds era5 --upload
```

code will crash because you overwrite `local_path` on line `L96` first and then it jumps right into `else` on `L110`